src/hash_object.cpp

```diff
@@ -3,23 +3,27 @@
 #include "tm/recursion_guard.hpp"
 #include "tm/vector.hpp"
 
-namespace Natalie {
+namespace TM {
 
 // this is used by the hashmap library and assumes that obj->env has been set
-size_t HashObject::hash(Key *&key) {
+size_t HashKeyHandler<Natalie::HashKey *>::hash(Natalie::HashKey *key) {
```

Comment:
I think the comment above this line is inaccurate (and was already inaccurate before this change). You can just remove the comment, if you want, because I think the code is probably clear enough now.